lib/kafka_ex/protocol/sync_group.ex

```diff
@@ -0,0 +1,66 @@
+defmodule KafkaEx.Protocol.SyncGroup do
+  import KafkaEx.Protocol.Common
+
+  @member_assignment_version 0
+
+  defmodule Response do
+    defstruct error_code: nil, assignments: []
+    @type t :: %Response{error_code: atom | integer, assignments: [{binary, [integer]}]}
```

Comment:
Can `error_code` be a generic atom or can we narrow it down to `nil | integer`?

Also, it might help to create a type for `assignment`.  I might even go so far as to create types for the `binary` and `integer` components of an assignment - i.e., what _are_ they?  I think it helps when trying to digest the code at a glance.  It can also generate helpful dialyzer warnings.